### PR TITLE
[ML] Better error when aggregate_metric_double used in scrolling data…

### LINKED ISSUE #90592: [ML] Investigate how aggregate_metric_double fields affect datafeeds
Since #87849 and #88534 it is possible that anomaly detection jobs and datafeeds will encounter the `aggregate_metric_double` type.

We need to investigate how this will affect certain aspects of their behaviour:

1. Specifying detector fields when the datafeed is using an aggregation
2. Specifying detector fields when the datafeed is not using an aggregation
3. Ability to switch a datafeed between using an aggregation and not using an aggregation without deleting and recreating the job
4. Correct identification of whether to use `_source` or doc values to get values from `aggregate_metric_double` fields
5. How to appropriately set `summary_count_field` when using `aggregate_metric_double`

Based on this investigation decide what changes we should make to either code or documentation.

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
…feed

Fields of type `aggregate_metric_double` cannot be used with anomaly detection datafeeds unless they are used in aggregations. This commit improves the error message in this scenario so that the user gets better informed of why they cannot use such fields.

Closes #90592

### CODE REVIEW

File: docs/changelog/92232.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+pr: 92232
+summary: Better error when `aggregate_metric_double` used in scrolling data…
```
Comment (droberts195) [inline docs/changelog/92232.yaml:2]:
```suggestion
summary: Better error when `aggregate_metric_double` used in scrolling datafeeds
```

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/scroll/ScrollDataExtractorFactory.java [hunk 2]
```diff
@@ -104,6 +111,17 @@ public static void create(
                 );
                 return;
             }
+            Optional<String> optionalAggregatedMetricDouble = findFirstAggregatedMetricDoubleField(fieldCapabilitiesResponse);
+            if (optionalAggregatedMetricDouble.isPresent()) {
+                listener.onFailure(
+                    ExceptionsHelper.badRequestException(
+                        "field [{}] is of type [{}] and cannot be used in datafeed without aggregations",
```
Comment (droberts195) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/scroll/ScrollDataExtractorFactory.java:118]:
```suggestion
                        "field [{}] is of type [{}] and cannot be used in a datafeed without aggregations",
```

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java [hunk 3]
```diff
@@ -828,4 +830,55 @@ public void testStartDatafeed_GivenTimeout_Returns408() throws Exception {
...
+
+        ElasticsearchStatusException e = expectThrows(
+            ElasticsearchStatusException.class,
+            () -> startDatafeed(datafeedConfig.getId(), 0L, null)
+        );
+
+        assertThat(e.status(), equalTo(RestStatus.BAD_REQUEST));
+        assertThat(
+            e.getMessage(),
+            containsString("field [presum] is of type [aggregate_metric_double] and cannot be used in datafeed without aggregations")
```
Comment (droberts195) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java:881]:
```suggestion
            containsString("field [presum] is of type [aggregate_metric_double] and cannot be used in a datafeed without aggregations")
```

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by dimitris-athanasiou** [comment]
@elasticmachine update branch

**Pull request merged**